### ISSUE TITLE
sv_privateclients does not require sv_privatePassword to be set

### ISSUE THREAD

**Issue by illwieckz** [DaemonEngine/Daemon#41]
I just spawn a server with `sv_maxclients` set to `4` and `sv_privateclients` set to `2` and 4 non-private clients can join.

**Comment by illwieckz** [comment]
easy to reproduce:

```sh
./daemonded -homepath /tmp/unvded -set sv_maxclients 1 \
    -set sv_privateclients 1 -set sv_privatePassword "53CR37" +devmap vega
```

```sh
./daemon -homepath /tmp/unvplay +connect localhost
```

The client must not be able to join without password (there is no non-private slot available), but it does.

**Comment by illwieckz** [comment]
The check is done there:

https://github.com/DaemonEngine/Daemon/blob/12254d88b583b41854ec43d68792f04f5c05950b/src/engine/server/sv_client.cpp#L131-L179

**Comment by slipher** [comment]
I think the most straightforward solution would be to have password validation always fail if the password is an empty string.

**Comment by illwieckz** [comment]
And by the way, it looks like this code is just making every server reading password from every player. Would better if server send a public key to client and receives from client the password encoded with it… Would not prevent mitm but would prevent anyone spamming password to anyone…

**Comment by illwieckz** [comment]
Well, even having a public/private key mechanism will not prevent the client to send the pass to any server… Except if the server keeps its private key forever and the client does not store the passwd but the encrypted one, sending to any server some blob only one server can read.

**Comment by mbasaglia** [comment]
Could use the same challenge system I've implemented for rcon:
* client sends a challenge request
* server responds with a random string
* client encrypts the random string with the password and sends that
* server checks if it matches with what it expects

**Comment by illwieckz** [comment]
or something like that, but whatever, something more secure than the _thing_ we have currently, I guess the g_password stuff (to deny connexion to non allowed people) is like this one.

**Comment by illwieckz** [comment]
I reproduced with wrong password:

```sh
./daemonded -homepath /tmp/unvded -set sv_maxclients 1 \
    -set sv_privateclients 1 -set sv_privatePassword "53CR37" +devmap vega
```

```sh
./daemon -homepath /tmp/unvplay -set password "WR0NG" +connect localhost
```

Note that I also tested the `+set` options if they was expected as `+set` ones…

**Comment by Viech** [comment]
Closing as merged PR #54 says it is fixing this issue.
